src/sentry/preprod/models.py

```diff
@@ -144,21 +144,41 @@ def as_choices(cls) -> tuple[tuple[int, str], ...]:
 
     def get_sibling_artifacts_for_commit(self) -> models.QuerySet[PreprodArtifact]:
         """
-        Get all artifacts for the same commit comparison (monorepo scenario).
+        Get sibling artifacts for the same commit, deduplicated by app_id.
 
-        Note: Always includes the calling artifact itself along with any siblings.
-        Results are filtered by the current artifact's organization for security.
+        When multiple artifacts exist for the same app_id (e.g., due to reprocessing),
+        this method returns only one artifact per app_id to prevent duplicate rows
+        in status checks:
+        - For the calling artifact's app_id: Returns the calling artifact itself
+        - For other app_ids: Returns the earliest (oldest) artifact for that app_id
+
+        Note: Results are filtered by the current artifact's organization for security.
 
         Returns:
-            QuerySet of PreprodArtifact objects, ordered by app_id for stable results
+            QuerySet of PreprodArtifact objects, deduplicated by app_id, ordered by app_id
         """
+        from collections import defaultdict
```

Comment:
nit: AI loves to do this for some reason but keep all imports at the top unless necessary to break import loops